cartridges/int_algolia/cartridge/scripts/algolia/lib/algoliaSplitter.js

```diff
@@ -61,11 +61,19 @@ function splitHtmlContent(htmlContent, maxByteSize, splitterElement) {
  * @returns {string} Sanitized content
  */
 function removeIgnoredContent(content) {
-    //remove restricted tags and their content
-    IGNORED_TAGS.forEach(function(tag) {
-        content = content.replace(new RegExp('<' + tag + '.*?' + tag + '>', 'g'), '');
-    });
-
+    for (let i = 0; i < IGNORED_TAGS.length; i++) {
+        let tag = IGNORED_TAGS[i];
+        let startIndex = content.indexOf('<' + tag);
```

Comment:
I see that you take the index of `<tag` and not `<tag>` to match self closing tags I imagine.
But you don't look for self closing tags for the `endIndex` so I fear that it won't work for those tags.
Could you check that with a unit test?

Reply:
You have added tests that call `splitHtmlContent` but haven't addressed the issue I was mentioning. Please add unit tests on `removeIgnoredContent`, you will see that it doesn't work has expected, it's the `removeHtmlTagsAndFormat` function that catch them later.